src/components/select/select-list/select-list.component.js

```diff
@@ -0,0 +1,153 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  useCallback
+} from 'react';
+import I18n from 'i18n-js';
+import PropTypes from 'prop-types';
+import StyledSelectList from './select-list.style';
+import Events from '../../../utils/helpers/events/events';
+import updateListScrollTop from './update-list-scroll';
+import filterChildren from '../../../utils/filter-children';
+import StyledOption from '../option/option.style';
+import highlightPartOfText from './highlight-part-of-text';
+
+const SelectList = ({
+  id,
+  children,
+  onSelectOption,
+  onSelectListClose,
+  filterText,
+  hasTypeToSearch
+}) => {
+  const list = useRef();
+  const [currentOptionsListIndex, setCurrentOptionsListIndex] = useState(-1);
+
+  const handleGlobalKeydown = useCallback((event) => {
+    let newIndex;
+    const isNoOptionSelected = currentOptionsListIndex === -1;
+    const optionsList = React.Children.toArray(children);
+    const lastIndex = optionsList.length - 1;
+
+    if (Events.isTabKey(event) || Events.isEscKey(event)) {
+      onSelectListClose();
+    } else if (Events.isEnterKey(event)) {
+      if (isNoOptionSelected) {
+        return;
+      }
+
+      const { text, value } = optionsList[currentOptionsListIndex].props;
+
+      onSelectOption({ text, value });
```

Comment:
This may be undefined if no `onSelectOption` prop has been passed in

Reply:
I'll set the `onSelectOption` prop as `isRequired`, this component is only presentational and is not designed to work outside of select components

Reply:
Yeah I think that is the better approach